app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ViewAttachmentController.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc._
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.models.Permission
+import uk.gov.hmrc.tariffclassificationfrontend.models.response.{FileMetadata, ScanStatus}
+import uk.gov.hmrc.tariffclassificationfrontend.service.FileStoreService
+import uk.gov.hmrc.tariffclassificationfrontend.views
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+@Singleton
+class ViewAttachmentController @Inject()(verify: RequestActions,
+                                         fileService: FileStoreService,
+                                         val messagesApi: MessagesApi,
+                                         implicit val appConfig: AppConfig) extends FrontendController with I18nSupport {
+
+  def get(id: String): Action[AnyContent] = (verify.authenticated andThen verify.mustHave(Permission.VIEW_CASES)).async
+  { implicit request: Request[AnyContent] =>
+    fileService.getFileMetadata(id) map {
```

Comment:
I would  add **case _**  in case you get any unexpected answer
Just as a preference  
      case Some(file) if isFileReady(file) => Redirect(Call(method = "GET", url = fileSubmitted.url.get))
      case Some(file) => Ok(views.html.view_attachment_unavailable(file))
      case _ => unexpected_failure_view

Reply:
What would the "unexpected failure view" show other than the existing "something has gone wrong" error that you would get when the match fails?